micrometer-observation/src/main/java/io/micronaut/micrometer/observation/DefaultObservedFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.micrometer.observation;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.observation.DefaultMeterObservationHandler;
+import io.micrometer.observation.GlobalObservationConvention;
+import io.micrometer.observation.ObservationFilter;
+import io.micrometer.observation.ObservationHandler;
+import io.micrometer.observation.ObservationPredicate;
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.tracing.Tracer;
+import io.micrometer.tracing.handler.TracingAwareMeterObservationHandler;
+import io.micrometer.tracing.propagation.Propagator;
+import io.micrometer.tracing.handler.DefaultTracingObservationHandler;
+import io.micrometer.tracing.handler.PropagatingReceiverTracingObservationHandler;
+import io.micrometer.tracing.handler.PropagatingSenderTracingObservationHandler;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Factory for Micrometer Observation integration. Creates the following beans:
+ *
+ * <ul>
+ *     <li>{@link ObservationRegistry} - Instrumented ObservationRegistry</li>
+ *     <li>{@link ObservationHandler} - Observation Handlers for exporting metrics, traces</li>
+ *     <li>{@link ObservationFilter} - ObservationFilter that adds common low cardinality key values to all observations</li>
+ * </ul>
+ */
+@Factory
+@Internal
+public final class DefaultObservedFactory {
+    @Singleton
+    ObservationRegistry observationRegistry(
+        List<ObservationPredicate> observationPredicates,
+        List<GlobalObservationConvention<?>> observationConventions,
+        List<ObservationHandler<?>> observationHandlers,
+        List<ObservationFilter> observationFilters
+    ) {
+        ObservationRegistry observationRegistry = ObservationRegistry.create();
+        observationHandlers.forEach(observationRegistry.observationConfig()::observationHandler);
```

Comment:
I don't think this is enough. You should group all `TracingObservationHandlers` together with a `FirstMatchingCompositeHandler`. Same for `MeterObservationHandler`. Check this out https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfiguration.java#L85-L107